### PR TITLE
Upgrade from clang 5.0 to 7.0 + Add GCC 8

### PR DESCRIPTION
**Pull request by nerzhul**
This permits to have more features in clang-format + up-to-date clang.

See also why we need to set a new clang-format variable: https://travis-ci.org/minetest/minetest/jobs/357587622

### CODE REVIEW

File: util/travis/common.sh [hunk 1]
```diff
@@ -13,9 +13,10 @@ set_linux_compiler_env() {
 	elif [[ "${COMPILER}" == "clang-3.6" ]]; then
 		export CC=clang-3.6
 		export CXX=clang++-3.6
-	elif [[ "${COMPILER}" == "clang-5.0" ]]; then
-		export CC=clang-5.0
-		export CXX=clang++-5.0
+	elif [[ "${COMPILER}" == "clang-6.0" ]]; then
+		export CC=clang-6.0
+		export CXX=clang++-6.0
+		export CXXFLAGS="-stdlib=libc++"
```
Comment (sfan5) [inline util/travis/common.sh]: why are you installing `libstdc++-5` when `libc++` is used here?

**Comment by nerzhul** [comment]
Currently build doesn't pass i'm trying to fix it

**Comment by sfan5** [comment]
Oh sorry didn't see.

**Comment by nerzhul** [comment]
rebase done + clang tidy update too

**Comment by nerzhul** [comment]
@sfan5 please re-review, now it's working properly :)

File: .travis.yml [hunk 2]
```diff
@@ -71,6 +73,15 @@ matrix:
           packages: ['gcc-7', 'g++-7']
           sources: &sources
             - ubuntu-toolchain-r-test
+            
+    - env: PLATFORM=Unix COMPILER=gcc-8
+      compiler: gcc
+      os: linux
+      addons:
```
Comment (sfan5) [inline .travis.yml]: gcc-7 could be replaced with gcc-8. I don't think we need to cover many/all compiler versions.

File: util/travis/clangtidy.sh [hunk 3]
```diff
@@ -5,6 +5,8 @@ needs_compile || exit 0
 
 if hash clang-tidy-5.0 2>/dev/null; then
 	CLANG_TIDY=clang-tidy-5.0
```
Comment (sfan5) [inline util/travis/clangtidy.sh]: should this be removed? (same applies to ` util/travis/lint.sh`)

Comment (nerzhul) [reply util/travis/clangtidy.sh]: exact

File: .travis.yml (hunk 2 above)
Comment (nerzhul) [reply .travis.yml]: okay

**Comment by nerzhul** [comment]
@sfan5 comments are fixed. I'm trying to make the link between scripts & travis more simple

**Pull request merged**